go/arrow/compute/utils.go

```diff
@@ -0,0 +1,83 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compute
```

Comment:
ssmall nit, this seems orthogonal to compute (it seems like a general buffer class?)

Reply:
it is a general buffer class, I wanted to ensure performance for the serialize/deserialize of the expressions with the ipc handler so I needed a buffer that could meet the io.Writer and io.Seeker interfaces efficiently and generate a `memory.Buffer` without extra copies of the data. That's why it's not exported from this package (doesn't start with an uppercase letter)